packages/@lwc/style-compiler/src/serialize.ts

```diff
@@ -106,6 +106,9 @@ export default function serialize(result: Result, config: Config): string {
             buffer += `${STYLESHEET_IDENTIFIER}.${KEY__SCOPED_CSS} = true;\n`;
         }
 
+        // register the stylesheet
+        buffer += `registerStylesheet(${STYLESHEET_IDENTIFIER});\n`;
```

Comment:
Because we export all the stylesheets including the imported ones (line 118), we leave this on a separate line so we're only registering the stylesheet we compiled here.